core/src/main/java/org/elasticsearch/search/lookup/LeafDocLookup.java

```diff
@@ -67,20 +67,20 @@ public void setDocument(int docId) {
     }
 
     @Override
-    public Object get(Object key) {
+    public ScriptDocValues<?> get(Object key) {
         // assume its a string...
         String fieldName = key.toString();
```

Comment:
Hmm - I think, yeah, I'll stop calling toString.

I didn't notice this comment and have already merged. I'll finish the merge and open a followup and ping you on it.